src/tscore/ink_sock.cc

```diff
@@ -111,6 +112,26 @@ safe_clr_fl(int fd, int arg)
   return flags;
 }
 
+int
+safe_write(int fd, const char *buffer, int len)
+{
+  int num_written = 0;
+  while (num_written < len) {
```

Comment:
I think this should be re-ordered such that the `write` is attempted first, and only if that doesn't fully succeed should `write_ready` be invoked. That reduces the impact of calling this for the (by far) common case.

Reply:
Good point. Thanks Alan.